shop/src/main/java/ro/msg/learning/shop/service/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,124 @@
+package ro.msg.learning.shop.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.ProductCategory;
+import ro.msg.learning.shop.domain.key.StockId;
+import ro.msg.learning.shop.dto.ProductCategoryDto;
+import ro.msg.learning.shop.dto.ProductDto;
+import ro.msg.learning.shop.dto.StockDto;
+import ro.msg.learning.shop.exception.LocationNotFoundException;
+import ro.msg.learning.shop.exception.ProductCategoryNotFoundException;
+import ro.msg.learning.shop.exception.ProductNotFoundException;
+import ro.msg.learning.shop.mapper.ProductCategoryMapper;
+import ro.msg.learning.shop.mapper.ProductMapper;
+import ro.msg.learning.shop.mapper.StockMapper;
+import ro.msg.learning.shop.message.Messages;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductCategoryRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.ProductService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static ro.msg.learning.shop.message.Messages.*;
+
+@Service
+public class ProductServiceImpl implements ProductService {
+    @Autowired
+    ProductRepository productRepository;
+    @Autowired
+    ProductCategoryRepository productCategoryRepository;
+    @Autowired
+    StockRepository stockRepository;
+    @Autowired
+    LocationRepository locationRepository;
+
+   @Autowired
```

Comment:
Format the code please. Keep it consistent! There are some shortcuts to format the code. Please check and use them.